src/frontend/planner_test/tests/testdata/append_only.yaml

```diff
@@ -13,13 +13,12 @@
     create table t2 (v1 int, v3 int) with (appendonly = true);
     select t1.v1 as id, v2, v3 from t1 join t2 on t1.v1=t2.v1;
   stream_plan: |
-    StreamMaterialize { columns: [id, v2, v3, t1._row_id(hidden), t2._row_id(hidden)], pk_columns: [t1._row_id, t2._row_id] }
-      StreamExchange { dist: HashShard(t1._row_id, t2._row_id) }
-        StreamAppendOnlyHashJoin { type: Inner, predicate: t1.v1 = t2.v1, append_only: true, output: [t1.v1, t1.v2, t2.v3, t1._row_id, t2._row_id] }
-          StreamExchange { dist: HashShard(t1.v1) }
-            StreamTableScan { table: t1, columns: [t1.v1, t1.v2, t1._row_id], pk: [t1._row_id], distribution: UpstreamHashShard(t1._row_id) }
-          StreamExchange { dist: HashShard(t2.v1) }
-            StreamTableScan { table: t2, columns: [t2.v1, t2.v3, t2._row_id], pk: [t2._row_id], distribution: UpstreamHashShard(t2._row_id) }
+    StreamMaterialize { columns: [id, v2, v3, t1._row_id(hidden), t2._row_id(hidden), t2.v1(hidden)], pk_columns: [t1._row_id, t2._row_id, id, t2.v1] }
+      StreamAppendOnlyHashJoin { type: Inner, predicate: t1.v1 = t2.v1, append_only: true, output: [t1.v1, t1.v2, t2.v3, t1._row_id, t2._row_id, t2.v1] }
```

Comment:
IIUC append-only joins do not have that problem. Maybe should be skipped.

Reply:
Emmm. Why you mark this as resolved?

Reply:
err... I think it is explained by another comment
> Yes, now we just use logical_pk as the stream_key and add rewrite the plan tree from bottom to maintain every plan node has logical_pk in the locical phase.
But in future we will move the rewriting phase on the stream plan tree to maintain every plan node has stream_key. so that the stream_key can refer to distribution key, append_only and other stream properties

Reply:
Oh I see